### DISCUSSION TITLE
Android Advertising

### DISCUSSION THREAD

**Discussion by MeloiMM** [shinyorg/shiny#683] [Q&A] [answered]
What data can I advertise on Android and what data can I receive from this advertisement as I scan from another Android device?
In the documentation, it says that I could get the Service UUID from scanning an advertisement. However, I could not get any Service UUID through scanning. So what I did is try to connect before getting the Service UUID and I did get it successfully by using the peripheral.Connect() then .GetServices().
But what I really want to try is to get the Service UUID by just advertising it from another device and scan it with another device (without connecting and getting services using the .Connect() and .GetServices).
So my real question is, **is it really possible to get a Service UUID by just advertising and scanning it**? because in the documentation, it says that it could advertise service UUID.
Anyways, I don't think it is an issue. I just want to confirm that my understanding of the documentation is correct that we can advertise Service UUID with an android device and scan it from another device.
Thanks!

**Comment by aritchie** [comment, marked as answer]
You can advertise a service UUID via the BLE hosting.  I would recommend looking through the source code at things like this since the BLE hosting module is quite small

**Reply by MeloiMM** [reply]
Yes thanks. I just found out that it is possible to advertise Service UUID but we should use proper format. Thanks!
